app/src/main/java/com/github/se/eduverse/ui/todo/TodoListScreen.kt

```diff
@@ -0,0 +1,164 @@
+                onClick = { showCompleted = false },
+                enabled = showCompleted,
+                colors =
+                    ButtonColors(Color.White, Color(0xFF217384), Color(0xFF217384), Color.White),
+                modifier = Modifier.testTag("currentTasksButton")) {
+                  Text("Current Tasks", modifier = Modifier.padding(8.dp), fontSize = 20.sp)
+                }
+            TextButton(
+                onClick = { showCompleted = true },
+                enabled = !showCompleted,
+                colors =
+                    ButtonColors(Color.White, Color(0xFF217384), Color(0xFF217384), Color.White),
+                modifier = Modifier.testTag("completedTasksButton")) {
+                  Text("Completed Tasks", modifier = Modifier.padding(8.dp), fontSize = 20.sp)
+                }
+          }
+          LazyColumn(modifier = Modifier.padding(16.dp)) {
+            val todosToShow = if (showCompleted) doneTodos.value else actualTodos.value
+            items(todosToShow.size) { i ->
+              val todo = todosToShow[i]
+              TodoItem(
+                  todo,
+                  { todoListViewModel.setTodoActual(it) },
+                  { todoListViewModel.setTodoDone(it) })
+            }
+          }
+        }
+      })
+}
+
+/**
+ * Composable that represents a single todo item
+ *
+ * @param todo the todo to display
+ * @param onUndo code executed when the done button is clicked while the todo is completed
+ * @param onDone code executed when the done button is clicked while the todo is not completed
+ */
+@Composable
+fun TodoItem(todo: Todo, onUndo: (Todo) -> Unit, onDone: (Todo) -> Unit) {
+  val completed = todo.status == TodoStatus.DONE
+  Card(
+      modifier = Modifier.testTag("todoItem_${todo.uid}").fillMaxWidth().padding(vertical = 4.dp),
+      shape = RoundedCornerShape(16.dp)) {
+        Row(verticalAlignment = Alignment.CenterVertically) {
+          IconButton(
+              onClick = {
+                if (completed) {
+                  onUndo(todo)
+                } else {
+                  onDone(todo)
+                }
+              },
+              modifier = Modifier.testTag("todoDoneButton_${todo.uid}").padding(8.dp).size(24.dp),
+              colors =
+                  IconButtonColors(
+                      if (completed) Color.Green else Color.LightGray,
+                      Color.White,
+                      Color.LightGray,
+                      Color.White)) {
+                Icon(
+                    Icons.Rounded.Done,
+                    contentDescription = "Todo Done",
+                    modifier = Modifier.size(16.dp))
+              }
+          Text(
+              todo.name,
+              modifier = Modifier.weight(1f).padding(start = 8.dp, end = 8.dp),
+              textDecoration = if (completed) TextDecoration.LineThrough else TextDecoration.None,
+              color = if (completed) Color.Gray else Color.Black)
```

Comment:
If you take into account previous comment, putting [...] else Color.Unspecified should work

Reply:
I don't understand what you're suggesting here

Reply:
I mean that if you change the background to transparent, when the app is in dark mode, black text might not be readable. I think the unspecified color adapt to the background (I might be wrong), so it would solve that problem. But we can also wait unitl we implement the dark mode to change those kind of things

Reply:
I checked and setting the text color to Color.Unspecified is better suited to manage the dark and light mades themes, I'll change it